static/js/message_events.js

```diff
@@ -139,12 +139,16 @@ export function insert_new_messages(messages, sent_by_this_client) {
             );
         }
     } else {
-        // we're in the home view, so update its list
+        // User is not narrowed to a stream / topic / PM, so we just update the home message
+        // list which is always kept maintained.
         render_info = message_util.add_new_messages(messages, message_lists.home);
     }
 
     if (sent_by_this_client) {
-        const need_user_to_scroll = render_info && render_info.need_user_to_scroll;
+        const need_user_to_scroll =
+            render_info &&
+            render_info.need_user_to_scroll &&
+            narrow_state.is_message_feed_visible();
```

Comment:
Hmm, why not have this check be in the calculation of `need_user_to_scroll`? I think logically, the user doesn't need to scroll if the thing isn't visible...

Reply:
Or maybe we rename `render_info.need_user_to_scroll` to something like `render_info.scroll_position_conceals_new_message` or something like that that describes what part of the question is being checked there.

Reply:
Makes sense to do it at the point of calculation of `need_user_to_scroll` which I think is a good name.